tsl/test/sql/cagg_api.sql

```diff
@@ -0,0 +1,158 @@
+-- This file and its contents are licensed under the Timescale License.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-TIMESCALE for a copy of the license.
+
+\c :TEST_DBNAME :ROLE_SUPERUSER
+
+SET timezone TO CET;
+SET datestyle TO ISO;
+
+CREATE VIEW continuous_aggregates AS
+SELECT mat_hypertable_id AS materialization_id,
+       format('%I.%I', user_view_schema, user_view_name)::regclass AS continuous_aggregate
+  FROM _timescaledb_catalog.hypertable
+  JOIN _timescaledb_catalog.continuous_agg
+    ON hypertable.id = continuous_agg.mat_hypertable_id;
+
+CREATE TABLE hyper_ts (time timestamp NOT NULL, value float);
+CREATE TABLE hyper_tstz (time timestamptz NOT NULL, value float);
+CREATE TABLE hyper_multi (time timestamptz NOT NULL, device int, value float);
+CREATE TABLE hyper_no_cagg (time timestamptz NOT NULL, device int, value float);
+CREATE TABLE normal_ts(time timestamp NOT NULL, value float);
+
+SELECT * FROM create_hypertable('hyper_ts', 'time');
+SELECT * FROM create_hypertable('hyper_tstz', 'time');
+SELECT * FROM create_hypertable('hyper_no_cagg', 'time');
+SELECT * FROM create_hypertable('hyper_multi', 'time', 'device', 4);
+
+INSERT INTO hyper_ts
+  SELECT time, ceil(random() * 100)::int
+    FROM generate_series('2025-01-01'::timestamp,
+                         '2025-01-06', '1m') time;
+
+INSERT INTO hyper_tstz
+  SELECT time, ceil(random() * 100)::int
+    FROM generate_series('2025-01-01'::timestamptz,
+                         '2025-01-06', '1m') time;
+
+CREATE MATERIALIZED VIEW ts_temperature_1h
+  WITH  (timescaledb.continuous) AS
+SELECT time_bucket('1 hour', time), avg(value)
+  FROM hyper_ts
+GROUP BY 1;
+
+CREATE MATERIALIZED VIEW ts_temperature_15m
+  WITH  (timescaledb.continuous) AS
+SELECT time_bucket('15 minutes', time), avg(value)
+  FROM hyper_ts
+GROUP BY 1;
+
+CREATE MATERIALIZED VIEW tstz_temperature_1h
+  WITH  (timescaledb.continuous) AS
+SELECT time_bucket('1 hour', time), avg(value)
+  FROM hyper_tstz
+GROUP BY 1;
+
+CREATE MATERIALIZED VIEW tstz_temperature_15m
+  WITH  (timescaledb.continuous) AS
+SELECT time_bucket('15 minutes', time), avg(value)
+  FROM hyper_tstz
+GROUP BY 1;
+
+CREATE MATERIALIZED VIEW multi_temperature_15m
+  WITH  (timescaledb.continuous) AS
+SELECT time_bucket('15 minutes', time), avg(value)
+  FROM hyper_multi
+GROUP BY 1;
```

Comment:
On thing is missing here is testing hierarchical caggs.

Reply:
I think we need to take this separately.